src/sentry/integrations/gitlab/integration.py

```diff
@@ -298,22 +319,23 @@ def build_integration(self, state):
         secret = sha1_text("".join([hostname, state["installation_data"]["client_id"]]))
 
         integration = {
-            "name": group["full_name"],
+            "name": group.get("full_name", hostname),
             # Splice the gitlab host and project together to
             # act as unique link between a gitlab instance, group + sentry.
             # This value is embedded then in the webook token that we
             # give to gitlab to allow us to find the integration a hook came
             # from.
-            "external_id": u"{}:{}".format(hostname, group["id"]),
+            "external_id": u"{}:{}".format(hostname, group.get("id", "instance")),
             "metadata": {
-                "icon": group["avatar_url"],
+                "icon": group.get("avatar_url", None),
                 "instance": hostname,
-                "domain_name": u"{}/{}".format(hostname, group["full_path"]),
+                "domain_name": u"{}/{}".format(hostname, group.get("full_path", "")),
```

Comment:
Could we trim the trailing `/` off of this string?

```suggestion
                "domain_name": u"{}/{}".format(hostname, group.get("full_path", "")).rtrim("/"),
```

Reply:
Added a right-trim on `/`.